crawler/file_processing.py

```diff
@@ -1418,7 +1418,13 @@ def is_within_cq_range(range_min: Decimal, range_max: Decimal, num: Decimal128)
         Returns:
             bool - whether the value lies within range
         """
-        return range_min <= num.to_decimal() <= range_max
+        min_compare = range_min.compare(num.to_decimal())
+        is_within_min = min_compare != Decimal("1") and not min_compare.is_nan()
+
+        max_compare = range_max.compare(num.to_decimal())
+        is_within_max = max_compare != Decimal("-1") and not max_compare.is_nan()
```

Comment:
I would add brackets () to group the comparison order:
```suggestion
        is_within_max = (max_compare != Decimal("-1")) and (not max_compare.is_nan())
```